Project2015To2017/Reading/ProjectPropertiesReader.cs

```diff
@@ -161,32 +140,25 @@ private static (List<string>, List<string>) ReadConfigurationPlatformVariants(Pr
 			platformList.Sort();
 			return (configurationList, platformList);
 
-			string[] ParseFromProperty(string name) => project.PrimaryPropertyGroup
-				.Elements(project.XmlNamespace + name)
-				.FirstOrDefault()
+			string[] ParseFromProperty(string name) => project.Property(name)
 				?.Value
 				.Split(new[] {';'}, StringSplitOptions.RemoveEmptyEntries);
 		}
 
 		private static void ReadPropertyGroups(Project project)
 		{
-			var (conditional, unconditional) = project.ProjectDocument.Root
+			project.PropertyGroups = project.ProjectDocument.Root
 				.Elements(project.XmlNamespace + "PropertyGroup")
-				.Split(x => x.Attribute("Condition") != null);
+				.ToList();
 
-			if (unconditional.Count == 0)
+			try
 			{
-				throw new NotSupportedException(
-					"No unconditional property group found. Cannot determine important properties like target framework and others.");
+				var _ = project.PrimaryPropertyGroup();
 			}
-
-			project.AdditionalPropertyGroups = conditional;
-
-			project.PrimaryPropertyGroup = unconditional[0];
-
-			foreach (var child in unconditional.Skip(1).SelectMany(x => x.Elements()))
+			catch
```

Comment:
We should be able to do this without using try catch for normal flow, right?

Reply:
An exception is already thrown (`First` instead of `FirstOrDefault`), this is just to change the exception kind and message.

Reply:
Right, but in that case catch a more specific exception, these very generic catch statements might hide a lot more than you initially intended

Reply:
Am I to make changes in this PR or can we delay that to the next PR?

Reply:
Whatever is most convenient to you

Reply:
Next PR it is (unable to work on that now).